File: base.dagger/src/main/java/alvin/base/dagger/ApplicationComponent.java
package alvin.base.dagger;

import javax.inject.Singleton;

import dagger.Component;
import dagger.android.AndroidInjector;
import dagger.android.support.AndroidSupportInjectionModule;

@Singleton
@Component(modules = {
        AndroidSupportInjectionModule.class,
        ApplicationModule.class
})
interface ApplicationComponent extends AndroidInjector<Application> {

    @Component.Builder
    abstract class Builder extends AndroidInjector.Builder<Application> { }
}

File: base.kotlin/src/main/java/alvin/base/kotlin/ApplicationComponent.kt
package alvin.base.kotlin

import alvin.base.kotlin.dagger.DaggerModule
import alvin.base.kotlin.dbflow.DBFlowModule
import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule
import javax.inject.Singleton

@Singleton
@Component(modules = [
    AndroidSupportInjectionModule::class,
    ApplicationModule::class,
    DaggerModule::class,
    DBFlowModule::class
])
interface ApplicationComponent : AndroidInjector<Application> {

    @Component.Builder
    abstract class Builder : AndroidInjector.Builder<Application>()
}
